Publish zero if locked

BriceRenaudeau opened the pull request:
Hi,

I found that when a topic is locked, the mux just stops publishing. This means that our robot depends on another timeout to stop (the smoother in our case).
![twist_mux_lock_timout](https://user-images.githubusercontent.com/48433002/235676321-25c4c68f-9a0e-441e-826a-8a0b988b57cf.png)

So I made a small modification to publish zero twists when the topic that has priority is locked.

To do so, I moved the lock check in the topic callback. If the topic is locked, I replace the message with `geometry_msgs::msg::Twist()`. If this topic has priority, the zero twist value will be published.

Hence the robot starts to brake immediately.

BriceRenaudeau committed:
Update twist_mux.cpp

remove lock test in `hasPriority`

BriceRenaudeau committed:
Update twist_mux.hpp

Move getLockPriority to public

BriceRenaudeau committed:
Update topic_handle.hpp

Publish zero if topic has priority and is locked

doisyg commented:
This seems very useful. Can we have an estimate of when it will be merged ? @bmagyar and @efernandez, do you know who currently maintains this package ?

bmagyar commented:
hi guys, sorry for the delay on this

could you please add a feature flag for this and have it off by default? I'd rather not change default behaviour straight away

bmagyar committed:
Merge branch 'rolling' into publish_zero_if_locked

BriceRenaudeau commented:
I will make the modification to make it optional. 

What do you mean by feature flag?

BriceRenaudeau committed:
Update twist_mux.cpp

get `pub_zer_if_locked` param and propagate

BriceRenaudeau committed:
Update topic_handle.hpp

Add `pub_zero_if_locked` parameter ti TopicHandle class.

Publish zero vel only if param is true.

BriceRenaudeau committed:
Update twist_mux_topics.yaml

add parameter to config file

BriceRenaudeau commented:
Hi @bmagyar, 

I made the changes to make it optional. 

It may be good to change the documentation too.
